File: frontend/src/types/job.ts
export const JOB_STATUSES = [
  "wishlist",
  "applied",
  "in_process",
  "offer",
  "rejected",
  "withdrawn",
  "ghosted",
] as const

export type JobStatus = (typeof JOB_STATUSES)[number]

export const JOB_STATUS_LABELS: Record<JobStatus, string> = {
  wishlist: "Wishlist",
  applied: "Applied",
  in_process: "In process",
  offer: "Offer",
  rejected: "Rejected",
  withdrawn: "Withdrawn",
  ghosted: "Ghosted",
}

export type Job = {
  id: string
  companyId: string
  title: string
  portal: string | null
  portalApplicationId: string | null
  appliedAt: string | null
  status: JobStatus
  jobUrl: string | null
  notes: string | null
  createdAt: string
  updatedAt: string
}

export type JobInput = {
  companyId: string
  title: string
  portal?: string | null
  portalApplicationId?: string | null
  appliedAt?: string | null
  status: JobStatus
  jobUrl?: string | null
  notes?: string | null
}

File: backend/src/main/java/com/jat/jat/job/domain/Job.java
package com.jat.jat.job.domain;

import com.jat.jat.shared.domain.BaseAuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "jobs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Job extends BaseAuditableEntity {

    @Column(name = "company_id", nullable = false)
    private UUID companyId;

    @Column(nullable = false)
    private String title;

    private String portal;

    private String portalApplicationId;

    private Instant appliedAt;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private JobStatus status = JobStatus.wishlist;

    @Column(length = 2048)
    private String jobUrl;

    @Column(columnDefinition = "text")
    private String notes;

    @PrePersist
    void defaultStatus() {
        if (status == null) {
            status = JobStatus.wishlist;
        }
    }
}
